packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -54,6 +54,45 @@ typedef ReorderCallback = void Function(int oldIndex, int newIndex);
 /// All [children] must have a key.
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=3fB1mxOsqJE}
+///
+/// This sample shows by dragging the user can reorder the items of the list.
+/// list of Numbers will be used to display items in listview.
+/// [List.generate]it is constructor which  is used to form a list of values.
+/// [onReorder] This parameter is compulsory. It will be called when list child
+/// is moved to a new position.
+///
+/// {@tool dartpad --template=stateful_widget_scaffold}
+///
+/// ```dart
+/// List<String> _list = List.generate(5, (i) => "${i}");
+///
+/// Widget build(BuildContext context){
+///   return ReorderableListView(
+///     padding : const EdgeInsets.symmetric(horizontal:40),
+///     children: _list.map((String string) => ListTile(
+///       key: Key(_list[_list.indexOf(string)]),
+///       title: Text(string))).toList(),
+///     onReorder: (oldIndex, newIndex){
+///       String old = _list[oldIndex];
+///       if (oldIndex > newIndex){
+///         for(int i = oldIndex; i> newIndex; i--){
+///           _list[i]= _list[i-1];
+///         }
+///         _list[newIndex]= old;
+///       }
+///       else{
+///         for(int i = oldIndex; i< newIndex-1;i++){
+///           _list[i]= _list[i+1];
+///         }
+///         _list[newIndex -1] = old;
+///       }
+///       setState((){});
```

Comment:
I would use the same technique described in [ReorderCallback's API docs](https://api.flutter.dev/flutter/material/ReorderCallback.html):

```dart
      onReorder: (oldIndex, newIndex) {
        setState(() {
          if (oldIndex < newIndex) {
            // removing the item at oldIndex will shorten the list by 1.
            newIndex -= 1;
          }

          final element = _list.removeAt(oldIndex);
          _list.insert(newIndex, element);
        });
      },
```